### PR TITLE
rebuild SSO ASP.NET walkthrough

### PR DESCRIPTION
**Pull request by Rick-Kirkham**
The SSO API changed, so this had to be rewritten.

### CODE REVIEW

**Comment by o365devx** [comment]
Docs Build status updates of commit _[9099dcf](https://github.com/OfficeDev/office-js-docs-pr/commits/9099dcf1ed9a365c7f938bea4daa911982ecd890)_: 

### :warning: Validation status: warnings


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md) | :warning:Warning | | [Details](#user-content-acf84d20-46d5f1db)
[docs/images/SelectHost.JPG](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/images/SelectHost.JPG) | :white_check_mark:Succeeded | |

<a id="acf84d20-46d5f1db"></a>
### [docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md)
 - **Line 151**: **[Warning]** `````Invalid file link:(~/develop/concepts/browsers-used-by-office-web-add-ins.md).`````

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C4%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201912042233103089-1471%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=pb%2B9yMMObOqbjY8p5%2Bhj9dL3HoZnbuwRs2qXdxPZMec%3D&st=2019-12-04T22%3A29%3A21Z&se=2020-01-04T22%3A34%3A21Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[79c9464](https://github.com/OfficeDev/office-js-docs-pr/commits/79c9464cbeace81320291aa1066d700b9112a719)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md) | :white_check_mark:Succeeded | |
[docs/images/SelectHost.JPG](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/images/SelectHost.JPG) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C4%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201912042244461534-1471%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=96ki8fKRxaow98sUGYch4H4jY9cPXBF9MmuUwTDCP6Y%3D&st=2019-12-04T22%3A40%3A59Z&se=2020-01-04T22%3A45%3A59Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[70f93b0](https://github.com/OfficeDev/office-js-docs-pr/commits/70f93b094a27fec7f5ad20e544447bbe5f7a6b9c)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md) | :white_check_mark:Succeeded | |
[docs/images/SelectHost.JPG](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/images/SelectHost.JPG) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C5%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201912051851234652-1471%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=iK42kJjfcYOtx1eHks97hs22HnAMYCf%2BVHgzTFn%2BFdk%3D&st=2019-12-05T18%3A47%3A39Z&se=2020-01-05T18%3A52%3A39Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[e5410f6](https://github.com/OfficeDev/office-js-docs-pr/commits/e5410f6193c1f2b35b4b3ac70ff6445e05565b74)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md) | :white_check_mark:Succeeded | |
[docs/images/SelectHost.JPG](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/images/SelectHost.JPG) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C5%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201912051859507713-1471%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=NV7K6a8o5VES9vKE5Tf263vsYWe%2Fexiadgcr7RDln50%3D&st=2019-12-05T18%3A56%3A02Z&se=2020-01-05T19%3A01%3A02Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/develop/create-sso-office-add-ins-aspnet.md [hunk 1]
```diff
@@ -1,14 +1,595 @@
...
+
+1. On the same page, choose the **Grant admin consent for [tenant name]** button, and then select **Yes** for the confirmation that appears.
+
+## Configure the solution
+
+1. In **Visual Studio**, right-click the top node in **Solution Explorer** (the Solution node, not either of the project nodes), and then select **Set startup projects**.
+
+1. Under **Common Properties**, select **Startup Project**, and then **Multiple startup projects**. Ensure that the **Action** for both projects is set to **Start**, and that the project that ends in "...WebAPI" is listed first.
+
+1. Still in **Solution Explorer**, choose the **Office-Add-in-Microsoft-Graph-ASPNETWebAPI** project. In **Properties**, ensure **SSL Enabled** is **True**. Verify that the **SSL URL** is `http://localhost:44355/`.
```
Comment (AlexJerabek) [inline docs/develop/create-sso-office-add-ins-aspnet.md]: I got a little confused by right-clicking on the node and going to that Properties menu.

File: docs/develop/create-sso-office-add-ins-aspnet.md [hunk 2]
```diff
@@ -1,14 +1,595 @@
...
+
+1. In web.config, use the values that you copied in earlier. Set both the **ida:ClientID** and the **ida:Audience** to your **Application (client) ID**, and set **ida:Password** to your client secret.
```
Comment (AlexJerabek) [inline docs/develop/create-sso-office-add-ins-aspnet.md]: Nit: For me, the file was capitalized "Web.Config". I think it should also be in quotes to be consistent with your other file references.

File: docs/develop/create-sso-office-add-ins-aspnet.md [hunk 3]
```diff
@@ -1,14 +1,595 @@
...
+
+    > [!NOTE]
+    > The **Application (client) ID** is the "audience" value when other applications, such as the Office host application (e.g., PowerPoint, Word, Excel), seek authorized access to the application. It is also the "client ID" of the application when it, in turn, seeks authorized access to Microsoft Graph.
+
+1. In the add-in project, open the add-in manifest file “Office-Add-in-ASPNET-SSO.xml” and then scroll to the bottom of the file. Just above the end `</VersionOverrides>` tag, you'll find the following markup:
```
Comment (AlexJerabek) [inline docs/develop/create-sso-office-add-ins-aspnet.md]: Consider explicitly stating the project name here.

File: docs/develop/create-sso-office-add-ins-aspnet.md [hunk 4]
```diff
@@ -1,14 +1,595 @@
...
+        // TODO 8: Handle all other Azure AD errors.
+	}
+    ```
+
+1. Replace `TODO 7` with the following. Note that on rare occasions the bootstrap token is unexpired when Office validates it, but expires by the time it is sent to Azure AD for exchange. Azure AD will respond with error AADSTS500133. When this happens, the code  recalls the SSO API (but no more than once). This time Office returns a new unexpired bootstrap token.
+
+    ```javascript
+    if ((exceptionMessage.indexOf("AADSTS500133") !== -1)
+        && (retryGetAccessToken <= 0))
+    {
```
Comment (AlexJerabek) [inline docs/develop/create-sso-office-add-ins-aspnet.md]: Nit: Please put this `{` on the preceding line.

File: docs/develop/create-sso-office-add-ins-aspnet.md [hunk 5]
```diff
@@ -1,14 +1,595 @@
...
+        retryGetAccessToken++;
+        getGraphData();
+    }
+    ```
+
+1. Replace `TODO 8` with the following.
+
+    ```javascript
+    else
+    {
```
Comment (AlexJerabek) [inline docs/develop/create-sso-office-add-ins-aspnet.md]: Nit: Please put this `{` on the preceding line.

File: docs/develop/create-sso-office-add-ins-aspnet.md [hunk 6]
```diff
@@ -1,14 +1,595 @@
...
+      <Scopes>
+          <Scope>Files.Read.All</Scope>
+          <Scope>offline_access</Scope>
+          <Scope>openid</Scope>
+          <Scope>profile</Scope>
+      </Scopes>
+    </WebApplicationInfo>
+    ```
+
+1. Replace the placeholder “$application_GUID here$” *in both places* in the markup with the Application ID that you copied when you registered your add-in. The "$" signs are not part of the ID, so do not include them. This is the same ID you used in for the ClientID and Audience in the web.config.
```
Comment (AlexJerabek) [inline docs/develop/create-sso-office-add-ins-aspnet.md]: In later steps, there's an explicit reminder to save the file, but not for Web.Config or Office-Add-in-ASPNET-SSO.xml

**Comment by o365devx** [comment]
Docs Build status updates of commit _[81d3c90](https://github.com/OfficeDev/office-js-docs-pr/commits/81d3c909a15e9977ebd1cab0648a68f9ec00e8d4)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md) | :white_check_mark:Succeeded | |
[docs/images/SelectHost.JPG](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/images/SelectHost.JPG) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C5%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201912052028250589-1471%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=kYR3Sk3IWxIE2b%2Fv5%2FVnGXVg6poxNfZHojITmv5Fups%3D&st=2019-12-05T20%3A24%3A40Z&se=2020-01-05T20%3A29%3A40Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[819a123](https://github.com/OfficeDev/office-js-docs-pr/commits/819a1232f22d82bd01d77534d2ec3956e7f2657e)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md) | :white_check_mark:Succeeded | |
[docs/images/SelectHost.JPG](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/images/SelectHost.JPG) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C5%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201912052029121414-1471%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=AIHCkH6KaxYCQTTM8enrcA9%2F39gDQq8ieE9gqzVnVIk%3D&st=2019-12-05T20%3A25%3A55Z&se=2020-01-05T20%3A30%3A55Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

File: docs/develop/create-sso-office-add-ins-aspnet.md [hunk 7]
```diff
@@ -1,14 +1,595 @@
...
+
+1. In the **Add new item** dialog name the file **Startup.Auth.cs** and then click **Add**.
+
+1. Shorten the namespace name in the new file to `Office_Add_in_ASPNET_SSO_WebAPI`.
+
+1. Ensure that all of the following `using` statements are at the top of the file.
+
+    ```csharp
+    using Owin;
+    using System.IdentityModel.Tokens;
```
Comment (AlexJerabek) [inline docs/develop/create-sso-office-add-ins-aspnet.md]: This should be `Microsoft.IdentityModel.Tokens`, not `System`.

**Comment by Rick-Kirkham** [comment]
> Ran through the tutorial and got up to running the add-in. I never actually saw OneDrive files loaded, but that may be due to my demo tenant settings. I figure since I'm getting authentication prompts, everything's probably setup correctly.
> I left some notes from my experience following the instructions. Would you like me to do a copy edit too?

I don't think a copy edit is essential at this time. Could you raise the problem of not seeing any files on the repo at [Office-Add-in-ASPNET-SSO/issues](https://github.com/OfficeDev/Office-Add-in-ASPNET-SSO/issues)? I can work on it after the new year. BTW, do you have any files in that OneDrive account?

**Comment by o365devx** [comment]
Docs Build status updates of commit _[c0da6e3](https://github.com/OfficeDev/office-js-docs-pr/commits/c0da6e39a1dd170715ee3dba192ece2386993ebb)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md) | :white_check_mark:Succeeded | |
[docs/images/SelectHost.JPG](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/images/SelectHost.JPG) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C5%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201912052146235862-1471%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=cO%2FEO0yMYSoPvdhMn55dOPLTPZOswFs%2FcrWNq3EOfCE%3D&st=2019-12-05T21%3A42%3A42Z&se=2020-01-05T21%3A47%3A42Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Comment by o365devx** [comment]
Docs Build status updates of commit _[a8fd968](https://github.com/OfficeDev/office-js-docs-pr/commits/a8fd9687cb8a912adafacea64a9cc68f71ba7585)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/develop/create-sso-office-add-ins-aspnet.md](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/develop/create-sso-office-add-ins-aspnet.md) | :white_check_mark:Succeeded | |
[docs/images/SelectHost.JPG](https://github.com/OfficeDev/office-js-docs-pr/blob/sso-aspnet-walkthrough/docs/images/SelectHost.JPG) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2019%5C12%5C5%5C57856ef7-e8cf-54ca-0d12-df5f42d1e8bb%5CPullRequest%5C201912052210155305-1471%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=eXjDy2Jh65pT%2B3aqOtluYWEbkMT%2BSfvprz8n3zV9ehQ%3D&st=2019-12-05T22%3A06%3A30Z&se=2020-01-05T22%3A11%3A30Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**